apollo-router-core/src/request.rs

```diff
@@ -44,34 +45,81 @@ where
     <Option<T>>::deserialize(deserializer).map(|x| x.unwrap_or_default())
 }
 
+#[derive(Deserialize)]
+struct RequestMeta {
+    query: String,
+    operation_name: Option<String>,
+}
+
+impl Request {
+    pub fn from_bytes(b: Bytes) -> Result<Request, serde_json::error::Error> {
+        let mut value = Value::from_bytes(b)?;
+
+        let (variables, extensions) = match &mut value {
+            Value::Object(object) => (
+                match object.remove("variables") {
+                    Some(Value::Object(o)) => o,
+                    None => Object::default(),
```

Comment:
:yarn: Null is often considered like None in query planner I noticed.

Reply:
should be fixed by 62ccdc9